src/pip/_internal/commands/__init__.py

```diff
@@ -3,57 +3,117 @@
 """
 from __future__ import absolute_import
 
-from pip._internal.commands.completion import CompletionCommand
-from pip._internal.commands.configuration import ConfigurationCommand
-from pip._internal.commands.debug import DebugCommand
-from pip._internal.commands.download import DownloadCommand
-from pip._internal.commands.freeze import FreezeCommand
-from pip._internal.commands.hash import HashCommand
-from pip._internal.commands.help import HelpCommand
-from pip._internal.commands.list import ListCommand
-from pip._internal.commands.check import CheckCommand
-from pip._internal.commands.search import SearchCommand
-from pip._internal.commands.show import ShowCommand
-from pip._internal.commands.install import InstallCommand
-from pip._internal.commands.uninstall import UninstallCommand
-from pip._internal.commands.wheel import WheelCommand
+import importlib
 
 from pip._internal.utils.typing import MYPY_CHECK_RUNNING
 
 if MYPY_CHECK_RUNNING:
-    from typing import List, Type
+    from typing import Any, Dict, Iterable, List, Tuple
     from pip._internal.cli.base_command import Command
 
+
+commands_dict = {
```

Comment:
Could not this be an OrderedDict ? To prevent the need for `commands_order` list ?
I'd say it wasn't because it was created when Python 2.6 was still supported.

Reply:
Sure, I can make that change.

Reply:
Definitely a sensible change! It might be a good idea to check if there's a good first issue for this since I remember that I'd thought about making one but don't know if I ever did. 🙈